appservice/src/intTest/groovy/nu/yona/server/UserTest.groovy

```diff
@@ -77,11 +78,87 @@ class UserTest extends AbstractAppServiceIntegrationTest
 		def response = appService.deleteUser(john)
 
 		then:
-		response.status == 200
+		appService.assertResponseStatus(response, 200)
 		def getUserResponse = appService.getUser(john.url, false)
 		getUserResponse.status == 400 || getUserResponse.status == 404
 	}
 
+	def 'Create John Doe with an Android device'()
+	{
+		given:
+		def ts = timestamp
+
+		when:
+		def johnAsCreated = createJohnDoe(ts, "My S8", "ANDROID")
+
+		then:
+		def johnAfterNumberConfirmation = appService.confirmMobileNumber(
+				{
+					AppService.assertResponseStatusSuccess(it)
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices".size == 1
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices"[0].name == "My S8"
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices"[0].operatingSystem == "ANDROID"
+				}, johnAsCreated)
+
+		cleanup:
+		appService.deleteUser(johnAsCreated)
+	}
+
+	def 'Create John Doe with an iOS device'()
+	{
+		given:
+		def ts = timestamp
+
+		when:
+		def johnAsCreated = createJohnDoe(ts, "iPhone X", "IOS")
+
+		then:
+		def johnAfterNumberConfirmation = appService.confirmMobileNumber(
+				{
+					AppService.assertResponseStatusSuccess(it)
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices".size == 1
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices"[0].name == "iPhone X"
+					assert it.responseData._embedded."yona:devices"._embedded."yona:devices"[0].operatingSystem == "IOS"
+				}, johnAsCreated)
+
+		cleanup:
+		appService.deleteUser(johnAsCreated)
+	}
+
+	def 'Try to create John Doe with an unsupported device operating system'()
+	{
+		given:
+		def ts = timestamp
+
+		when:
+		def johnAsCreated = appService.addUser(
+				{
+					AppService.assertResponseStatus(it, 400)
+					assert it.responseData.code == "error.device.unknown.operating.system"
+				}, "John", "Doe", "JD",
+				makeMobileNumber(ts), "My Raspberry", "RASPBIAN")
+
+		then:
+		assert johnAsCreated == null // Creation failed
+	}
+
+	def 'Try to create John Doe with device operating system UNKNOWN'()
+	{
+		given:
+		def ts = timestamp
+
+		when:
+		def johnAsCreated = appService.addUser(
+				{
+					AppService.assertResponseStatus(it, 400)
+					assert it.responseData.code == "error.device.unknown.operating.system"
+				}, "John", "Doe", "JD",
+				makeMobileNumber(ts), "First device", "UNKNOWN")
+
+		then:
+		assert johnAsCreated == null // Creation failed
+	}
+
```

Comment:
Some other test suggestions: removing a device, adding a second device.

Reply:
Yes, but that's part of another ticket: [YD-497](http://jira.yona.nu/browse/YD-497)